### PR TITLE
enh: `slice`, `starts_with`, `tail` methods for `str` namespace

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [x] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

None

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

Implements `slice`, `starts_with` and `tail` methods in the str namespace for Expr and Series.

It is a quite large one, but slice implementation allowed to refactor head and tail, however using slice does not allow for negative indexes as polars does. Since they were not supported anyway in the first place, I moved to this implementation but happy to discuss how to work around it.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 site_name: Narwhals
 repo_url: https://github.com/narwhals-dev/narwhals.git
+watch:
+  - narwhals
```
Comment (MarcoGorelli) [inline mkdocs.yml:4]: ooh, what's this?

File: narwhals/series.py [hunk 2]
```diff
@@ -1810,9 +1961,53 @@ def head(self, n: int = 5) -> Series:
                "zukky"
             ]
         """
-        if self._series._is_polars:
-            return self._series.__class__(self._series._series.str.slice(0, n))
-        return self._series.__class__(self._series._series.str.head(n))
+        return self._series.__class__(self._series._series.str.slice(0, n))
```
Comment (MarcoGorelli) [inline narwhals/series.py:1964]: can we just call the underlying library directly here? so, keep it as it is, but in `_pandas_like`, you make `head` call `slice`

File: narwhals/series.py [hunk 3]
```diff
@@ -1810,9 +1961,53 @@ def head(self, n: int = 5) -> Series:
...
+            shape: (4,)
+            Series: '' [str]
+            [
+               "atata"
+               "taata"
+               "atata"
+               "kkyun"
+            ]
+        """
+        return self._series.__class__(self._series._series.str.slice(-n))
```
Comment (MarcoGorelli) [inline narwhals/series.py:2010]: same

**Review by MarcoGorelli — COMMENTED** [review]
awesome, thanks @FBruzzesi ! 🏆

File: mkdocs.yml (hunk 1 above)
Comment (FBruzzesi) [reply mkdocs.yml:4]: Sorry it slipped in, it is for dynamic reloading if something is in the folder changes

File: narwhals/series.py (hunk 2 above)
Comment (FBruzzesi) [reply narwhals/series.py:1964]: The min version of Polars we support does not have `str.head`. If you look at the diff, it was already implemented using slice for Polars.

I thought to implement `str.slice` for pandas as well and simplify the API.

WDYT?

Comment (FBruzzesi) [reply narwhals/series.py:1964]: I think the problem is more on the Polars side, that our min supported version does not implement `str.head`, hence we use slice.

I thought that implementing pandas `str.slice` would bring both a feature and unify the API call in here.

WDYT?

Comment (MarcoGorelli) [reply narwhals/series.py:1964]: aaah, I see, thanks, that's why we were using `.str[:n]` in the first place 😄 cool, looks fine then, thanks for explaining!

**Review by MarcoGorelli — APPROVED** [review]
nice one, feel free to merge!

**Pull request merged**